scripts/remove-unused-locales.ts

```diff
@@ -0,0 +1,204 @@
+    .readdirSync(path.join(__dirname, "../locales"), {
+      withFileTypes: true
+    })
+    .filter(d => d.isDirectory());
+  const localeNamesArr = localeFolders.map(d => d.name);
+  console.log(
+    chalk.green(
+      `found ${localeNamesArr.length} locales in the project: ${localeNamesArr}`
+    )
+  );
+  return localeNamesArr;
+};
+
+/**
+ *  self explanatory, uses yq to delete a key from a locale
+ * @param entry the key to delete
+ * @param localeName the locale to delete the key from,
+ * e.g. 'en' or 'fr'
+ */
+const deleteYmlEntry = (entry: string, localeName: string) =>
+  new Promise(res =>
+    exec(
+      `yq -i 'del(.${entry})' ${path.join(
+        path.join(__dirname, "../locales"),
+        localeName,
+        "index.yml"
+      )}`,
+      (err, _stdout, _) => {
+        if (err) {
+          console.log(chalk.red(`error deleting ${entry}`));
+          return res(false);
+        } else {
+          console.log(chalk.grey(`deleted ${entry}`));
+          return res(true);
+        }
+      }
+    )
+  );
+
+/**
+ * uses yq to look for null or undefined entries, skippable
+ * @param locale  the locale to clean up, e.g. 'en' or 'fr'
+ * @returns  a promise that resolves to true if the cleanup was successful
+ */
+const cleanupLocales = async (locale: string) =>
+  new Promise(res =>
+    exec(
+      `yq -i " del(.. |select(.==null or .== {})) " ` +
+        path.join(__dirname, "../locales", locale, "index.yml"),
+      (err, _stdout, _) => {
+        if (err) {
+          console.log(chalk.red(`${err} error cleaning up ${locale}`));
+          return res(false);
+        } else {
+          console.log(chalk.green(`cleaned up ${locale} locale`));
+          return res(true);
+        }
+      }
+    )
+  );
+
+const findUnused = async (root: string, locale: string) => {
+  const d = await readLocaleDoc(root, locale);
+  const keys = extractKeys(d.doc);
+  const unusedPromises = await Promise.all(keys.map(k => isUnused(k)));
+  return keys.filter((_, idx) => unusedPromises[idx]);
+};
+
+const main = async (root: string, locale: string) => {
+  const localeLangNames = getAllLocaleNames();
+  const unusedKeys = await findUnused(root, locale);
+
+  if (unusedKeys.length === 0) {
+    console.log(chalk.green("no unused keys found!"));
+    process.exit(1);
+  }
+  console.log(chalk.bgBlue("checking for yq installation..."));
+  await installYqIfNeeded();
+  console.log(chalk.red(`found ${unusedKeys.length} unused keys, deleting...`));
+  // for every locale, delete every unused key
+  for (const localeName of localeLangNames) {
+    console.log(
+      chalk.green(
+        `deleting unused keys from ${localeName.toUpperCase()} locale`
+      )
+    );
+    for (const k of unusedKeys) {
+      await deleteYmlEntry(k, localeName);
+    }
+    // do something to clean locale up, in order to avoid null or empty objects
+    await cleanupLocales(localeName);
+  }
+};
+
+main(path.join(__dirname, "../locales"), "en").then(
```

Comment:
Why we start looking for unused keys inside the `en` locale file and not inside the `it` one?

Reply:
it should have no effect: since `IT` is the primary locale and `EN` is the secondary one, both need to have the same keys or else the generation would fail.
Also, this is how they are originally checked in the "locales_unused" script, so I guessed it was all good like this 🤷

Reply:
I don't well remember but I think that now the "main" locale should be consider the `it` one. If we'll remove the `en` from the required ones like `de` then we should remember to update this script too. IMHO I'd use the `it` one as the base file non the less. Let's also see what @thisisjp think about that.

Reply:
We're currently overseeing only strings in Italian (while English is translated at best effort). The ideal workflow (which would allow us also to better identify what needs to be translated) consists of having Italian as "default" language (mandatory), and used as a fallback whenever a string in another language is missing.

I know this is not the current setup, but we should aim to achieve it sooner or later.

Reply:
```suggestion
main(path.join(__dirname, "../locales"), "it").then(
```